src/pages/index.js

```diff
@@ -19,32 +21,169 @@ const addCardButton = document.querySelector(".profile__add-button");
 const addCardModal = document.querySelector("#card-edit-modal");
 const addCardForm = addCardModal.querySelector(".modal__form");
 
+const avatarButton = document.querySelector(".profile__avatar-button");
+const avatarModal = document.querySelector("#avatar-edit-modal");
+const avatarForm = avatarModal.querySelector(".modal__form");
+
+const imagePopUp = new PopupWithImage({ popupSelector: "#preview-modal" });
+const popupDelete = new PopupDelete({ popupSelector: "#delete-card-modal" });
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "c5770e84-6b43-47e8-8ef9-37352d6891bc",
+    "Content-Type": "application/json",
+  },
+});
+
+const userInfo = new UserInfo({
+  nameSelector: ".profile__name",
+  descriptionSelector: ".profile__description",
+  userAvatar: ".profile__avatar",
+});
+
+let cardSection;
+
+api.getInitialCards().then((initialCards) => {
+  cardSection = new Section(
+    {
+      items: initialCards,
+      renderer: (data) => {
+        const cardElement = new Card(
+          data,
+          "#card-template",
+          handleImageClick,
+          handleDelete,
+          handleLikeCard
+        );
+        cardSection.addItem(cardElement.getCard());
+      },
+    },
+    ".cards__list"
+  );
+  cardSection.renderItems();
+});
+
+api.loadUserInfo().then((userData) => {
+  userInfo.setUserInfo({
+    name: userData.name,
+    about: userData.about,
+  });
+  userInfo.setAvatar({ avatar: userData.avatar });
+});
+
 /* functions */
 const handleImageClick = (data) => {
   imagePopUp.open(data);
 };
 
+function handleDelete(card) {
+  popupDelete.open();
+  popupDelete.setSubmitAction(() => {
+    api
+      .deleteInitialCards(card.id)
+      .then(() => {
+        card.handleDelete();
+        popupDelete.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  });
+}
+
+function handleLikeCard(card) {
+  if (!card.isLiked) {
+    api
+      .likeCard(card.id)
+      .then(() => {
+        card.isLiked = true;
```

Comment:
Needs correction: please don't change the card instance variables directly, create a setter. Also don't forget to remove `setIsLiked` method, you wan't need it anymore as you should change the button color depending on the isLiked state, not just toggle

```js
setIsLiked(isLiked) { 
   // set instance variable
    this._isLiked = isLiked; 
    this._handleLikeStatus(); 
} 
```